diff --git a/password_policy.module b/password_policy.module
index 8392f8b..21a4cfb 100644
--- a/password_policy.module
+++ b/password_policy.module
@@ -27,14 +27,31 @@ function password_policy_help($path, $arg) {
  * Implements hook_init().
  */
 function password_policy_init() {
-  global $user;
+	global $user;
+	
+	// Check password reset status and force a reset if needed.
+	if (_password_policy_is_password_change_forced($user->uid) &&
+			!_password_policy_is_path_allowed_when_password_change_forced()) {
+		if(_isHybridAuthUserOrAdmin($user)){
+			$user->force_password_change=0;
+		}else{
+			_password_policy_set_password_change_forced_message();
+			_password_policy_go_to_password_change_page();
+		}
+		
+	}
+}
 
-  // Check password reset status and force a reset if needed.
-  if (_password_policy_is_password_change_forced($user->uid) &&
-      !_password_policy_is_path_allowed_when_password_change_forced()) {
-    _password_policy_set_password_change_forced_message();
-    _password_policy_go_to_password_change_page();
-  }
+/**
+ * Check if current user is hybridauth user or admin
+ */
+function _isHybridAuthUserOrAdmin($user){
+	if(!empty(_hybridauth_identity_load_by_uid($user->uid)) || (array_search('administrator', $user->roles))){
+		return true;
+	}
+	else{
+		return false;
+	}
 }
 
 /**
@@ -376,7 +393,7 @@ function password_policy_user_login(&$edit, $account) {
   $roles = is_array($account->roles) ? array_keys($account->roles) : array();
   $policy = _password_policy_load_active_policy($roles);
   // A value $edit['name'] is NULL for a one time login.
-  if ($policy && ((!empty($account->uid) && $account->uid > 1) || variable_get('password_policy_admin', 1)) && !empty($edit['values']['name'])) {
+  if ($policy && ((!empty($account->uid) && $account->uid > 1) || variable_get('password_policy_admin', 1)) && !empty($edit['values']['name']) && !_isHybridAuthUserOrAdmin($account)) {
     // Calculate expiration and warning times.
     $expiration = $policy['expiration'];
     $warning = max(explode(',', $policy['warning']));
