diff --git a/password_policy.module b/password_policy.module
index 8392f8b..a968993 100644
--- a/password_policy.module
+++ b/password_policy.module
@@ -27,14 +27,31 @@ function password_policy_help($path, $arg) {
  * Implements hook_init().
  */
 function password_policy_init() {
-  global $user;
+	global $user;
+	
+	// Check password reset status and force a reset if needed.
+	if (_password_policy_is_password_change_forced($user->uid) &&
+			!_password_policy_is_path_allowed_when_password_change_forced()) {
+		if(_isHybridAuthUser($user)){
+			$user->force_password_change=0;
+		}else{
+			_password_policy_set_password_change_forced_message();
+			_password_policy_go_to_password_change_page();
+		}
+		
+	}
+}
 
-  // Check password reset status and force a reset if needed.
-  if (_password_policy_is_password_change_forced($user->uid) &&
-      !_password_policy_is_path_allowed_when_password_change_forced()) {
-    _password_policy_set_password_change_forced_message();
-    _password_policy_go_to_password_change_page();
-  }
+/**
+ * Check if current user is hybridauth user
+ */
+function _isHybridAuthUser($user){
+	if(!empty($user->data['hybridauth'])){
+		return true;
+	}
+	else{
+		return false;
+	}
 }
 
 /**
